File: spring-web-java/src/main/java/spring/web/java/infrastructure/persistence/RequestLockRedisRepository.java
package spring.web.java.infrastructure.persistence;

import java.time.Duration;

import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.stereotype.Repository;

import lombok.RequiredArgsConstructor;

@Repository
@RequiredArgsConstructor
public class RequestLockRedisRepository {

	private final RedisTemplate<String, String> redisTemplate;

	public boolean setIfAbsent(Long memberId, String method, String uri) {
		Boolean result = redisTemplate.opsForValue()
			.setIfAbsent("request_lock", String.format("%s:%s:%s", memberId, method, uri), Duration.ofSeconds(1));

		return Boolean.TRUE.equals(result);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/infrastructure/persistence/RequestLockRedisRepository.kt
package spring.web.kotlin.infrastructure.persistence

import org.springframework.data.redis.core.RedisTemplate
import org.springframework.stereotype.Repository
import java.time.Duration

@Repository
class RequestLockRedisRepository(
    private val redisTemplate: RedisTemplate<String, String>
) {
    fun setIfAbsent(memberId: Long, method: String, uri: String): Boolean {
        val result = redisTemplate.opsForValue()
            .setIfAbsent("request_lock", "$memberId:$method:$uri", Duration.ofSeconds(1))

        return result == true
    }
}